File: opensrp-app/src/main/java/org/smartregister/workmanager/jobs/BaseWorker.java
package org.smartregister.workmanager.jobs;

import android.content.Context;

import androidx.annotation.NonNull;
import androidx.work.Worker;
import androidx.work.WorkerParameters;

import org.jetbrains.annotations.NotNull;

public abstract class BaseWorker extends Worker {

    @NotNull
    public Context getContext() {
        return super.getApplicationContext();
    }

    public BaseWorker(@NonNull Context context, @NonNull WorkerParameters workerParams) {
        super(context, workerParams);
    }

    @Override
    public abstract Result doWork();
}

File: opensrp-core/src/main/java/org/smartregister/sync/wm/worker/BaseWorker.kt
package org.smartregister.sync.wm.worker

import android.content.Context
import androidx.work.Worker
import androidx.work.WorkerParameters
import org.smartregister.CoreLibrary
import org.smartregister.util.WorkerNotificationDelegate

abstract class BaseWorker(context: Context, workerParams: WorkerParameters) : Worker(context, workerParams){

    val notificationDelegate = WorkerNotificationDelegate(context, getTitle())

    fun beforeWork(){
        val coreLibrary = CoreLibrary.getInstance()
        val syncConfiguration = coreLibrary.syncConfiguration
        if (syncConfiguration != null){
            coreLibrary.context().httpAgent
                .apply {
                    connectTimeout = syncConfiguration.connectTimeout
                    readTimeout = syncConfiguration.readTimeout
                }
        }
    }

    abstract fun getTitle() : String

}